### PR TITLE
Customize pip-compile header

### LINKED ISSUE #41303: pip-compile and CUSTOM_COMPILE_COMMAND
### Environment

SaaS (https://sentry.io/)

### Version

_No response_

### Steps to Reproduce

JFYI

https://github.com/getsentry/sentry/blob/2e36d4d3a3926e9efe06287e59d27be1c40300fa/tools/freeze_requirements.py#L13

There is `CUSTOM_COMPILE_COMMAND` env to [customize](https://github.com/jazzband/pip-tools#configuration) the header:

```
$ CUSTOM_COMPILE_COMMAND="make freeze-requirements" pip-compile
#
# This file is autogenerated by pip-compile with python 3.11
# To update, run:
#
#    make freeze-requirements
#
....
```

### Expected Result

N/A

### Actual Result

N/A

### PR DESCRIPTION
**Pull request by atugushev**
Resolves #41303.

### Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. and is gonna need some rights from me in order to utilize my contributions in this here PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

### CODE REVIEW

File: tools/freeze_requirements.py [hunk 1]
```diff
@@ -10,26 +11,14 @@
...
-    # unfortunately rewrite the whole file.
-    dest = args[-1]
+    env = os.environ.copy()
+    env["CUSTOM_COMPILE_COMMAND"] = "make freeze-requirements"
+
     try:
-        run(args, check=True, capture_output=True)
+        run(args, check=True, capture_output=True, env=env)
     except CalledProcessError as e:
         raise e
```
Comment (asottile-sentry) [inline tools/freeze_requirements.py:21]: while we're here -- I think the `raise e` is leftover from a debugging check -- can remove the `try: except:` entirely

**Review by asottile-sentry — APPROVED** [review]
looks good!  can either ignore my comment or apply it

File: tools/freeze_requirements.py (hunk 1 above)
Comment (atugushev) [reply tools/freeze_requirements.py]: Done ac7d28b99d2f8f6f882242c6a844cca28c165a08

**Review by asottile-sentry — APPROVED** [review]

**Pull request merged**